dulwich/errors.py

```diff
@@ -84,6 +84,10 @@ class NotBlobError(WrongObjectException):
     type_name = "blob"
 
 
+class CheckoutError(Exception):
```

Comment:
Can you give this a less generic name? There are all kinds of errors that can occur during checkout.

Reply:
raising DirNotCleanError in the new commit